e2e/support/helpers/e2e-downloads-helpers.ts

```diff
@@ -107,37 +113,56 @@ export function downloadAndAssert(
     });
 }
 
-export function assertSheetRowsCount(expectedCount) {
-  return sheet => {
+export function assertSheetRowsCount(expectedCount: number) {
+  return (sheet: any) => {
     const range = xlsx.utils.decode_range(sheet["!ref"]);
     expect(range.e.r).to.eq(expectedCount);
   };
 }
 
-export function assertNotEmptyObject(data) {
+export function assertNotEmptyObject(data: any) {
   return Object.keys(data).length > 0;
 }
 
-function getEndpoint(
+type GetEndPointParams = Pick<
+  DownloadAndAssertParams,
+  "fileType" | "questionId" | "publicUuid" | "dashcardId" | "dashboardId"
+>;
+function getEndpoint({
   fileType,
   questionId,
   publicUuid,
   dashcardId,
   dashboardId,
-) {
+}: GetEndPointParams): {
+  endpoint: string;
+  method: "POST" | "GET";
+} {
   if (dashcardId != null && dashboardId != null) {
-    return `api/dashboard/${dashboardId}/dashcard/${dashcardId}/card/${questionId}/query/${fileType}**`;
+    return {
+      endpoint: `api/dashboard/${dashboardId}/dashcard/${dashcardId}/card/${questionId}/query/${fileType}**`,
```

Comment:
👍 I think this is easier to understand for now, but if later on, like you said, we turns all of these endpoints into the same method then we could just return the URL that time.